### PR TITLE
gh-reference: add page

### PR DESCRIPTION
**Pull request by navarroaxel**
- [x] The page (if new), does not already exist in the repo.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

For: #4701

### CODE REVIEW

File: pages/common/gh-reference.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# gh reference
+
+> Display a reference about GitHub CLI command.
```
Comment (bl-ue) [inline pages/common/gh-reference.md:3]:
```suggestion
> Display a reference about the GitHub CLI command.
```

**Review by bl-ue — CHANGES_REQUESTED** [review]
One tiny nit, otherwise LGTM!

**Review by bl-ue — APPROVED** [review]

**Comment by bl-ue** [comment]
This is a weird, but useful, command.  It's strange that it's listed in the `gh` help or tab-completion 🤔

**Comment by navarroaxel** [comment]
> This is a weird, but useful, command. It's strange that it's listed in the `gh` help or tab-completion thinking

Maybe because this is listed in the `gh help` section in the man pages, it's like an easter egg that this runs as a subcommand.

**Comment by bl-ue** [comment]
>`gh help` section in the man pages

For some reason or other I'm not getting this.  When I run `man gh` it doens't say anything about `gh help`.

**Comment by navarroaxel** [comment]
@bl-ue actually I was talking about the web man pages https://cli.github.com/manual/gh_help_reference

**Comment by bl-ue** [comment]
Oh oops!! 🤣

**Review by sbrl — APPROVED** [review]
Nice one!

**Pull request merged**